### PR TITLE
Add djoser package for password reset functionality

### PR DESCRIPTION
**Pull request by johngriebel**
feature/issue #266 - Backed Support for Password Reset.

Signed-off-by: John Griebel <jgriebel@3blades.io>

### CODE REVIEW

File: appdj/settings/base.py [hunk 1]
```diff
@@ -237,6 +238,14 @@
     'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.URLPathVersioning'
 }
 
+DJOSER = {
+    'DOMAIN': "",
```
Comment (johngriebel) [inline appdj/settings/base.py]: Djoser works by sending the user an email to the one provided by the `POST` request to the `auth/password/reset` endpoint. The email contains a link to a _frontend_ page specified by the `PASSWORD_RESET_CONFIRM_URL` setting below. This domain defaults to the `name` that corresponds to the database record referred to by `SITE_ID`. For us, this is currently example.com, which is clearly incorrect. I'm not sure what this should be; It's unclear to me whether or not the front end domain is already stored in an environment variable somewhere. @raygesualdo @jgwerner

File: actions/middleware.py [hunk 2]
```diff
@@ -125,7 +125,8 @@ def _get_object_pk_from_response_data(self, data) -> str:
 
     @staticmethod
     def _get_model_from_func(view_func):
-        if hasattr(view_func, 'cls') and hasattr(view_func.cls, 'queryset'):
+        if (hasattr(view_func, 'cls') and hasattr(view_func.cls, 'queryset')
+                and getattr(view_func.cls, 'queryset') is not None):
```
Comment (jkp85) [inline actions/middleware.py]: If you need to know if queryset is not None you can skip `hasattr` altogether, you can do:
```python
if hasattr(view_func, 'cls') and getattr(view_func.cls, 'queryset', None):
```

File: appdj/settings/base.py [hunk 3]
```diff
@@ -237,6 +238,14 @@
...
+    'SITE_NAME': "3Blades",
+    'PASSWORD_RESET_CONFIRM_URL': "password/reset/{uid}/{token}",
```
Comment (johngriebel) [inline appdj/settings/base.py]: There will have to be a front end page of the form `(dev|test|staging etc).3blades.ai/password/<uid>/<token>` that sends a `POST` request to the `auth/password/reset/confirm` endpoint with the following `POST` parameters: 
- `uid` - Note that this is _not_ `uuid`. This is important
- `token` - The token generated by Django for this password reset.
- `new_password` - The new password for the user.

Comment (raygesualdo) [inline appdj/settings/base.py]: Let's changes the front end URL from `password/reset/{uid}/{token}` to `/auth/password-reset?uid={uid}&token={token}` to match our current front end routing scheme.

**Pull request merged**